### DISCUSSION TITLE
Introduction of the project on main page

### DISCUSSION THREAD

**Discussion by Branikolog** [ihhub/fheroes2#3880] [Ideas]
Currently, our main page is lacking the description of fheroes2 project. People sometimes even don't know what is that, what is it for and what is the difference, comparing to the original game. I believe people would like to know main goals of our project too, as well as the list of features we've already implemented, which couldn't be found in the original game. 
![Page](https://user-images.githubusercontent.com/55348946/126498157-c8e11963-4d1d-42de-8ab2-903d04dff577.png)

**Comment by Branikolog** [comment]
So, I propose to add some text about the project on the main page.

Here's some sort of draft of this description: 
(Sorry for mistakes :) )
_Free Heroes of Might and Magic II (fheroes2) is a recreation of HoMM2 game engine. This open-source multi-platform project written with C++ from a scratch is aimed to fully replicate the original game with meaningful changes for gameplay, graphics and logic, bringing a new life to one of the most anticipated turn-based strategies. 
Final release will be a full faithful remake with high resolution support, improved AI, numerous UI fixes and improvements._ (Here we can place a link to a separate list of new features and fixes, which I provide in the further post.) 
_With the help of fheroes2 open engine new creatures, spells, game mechanics, skills and even castles could be added. Also improved video quality, new sounds and extended multiplayer can be implemented using the capabilities of the engine. Many of this and even more is expected to appear in the future expansion by our team called "Resurrection"._

**Comment by Branikolog** [comment]
It will be nice, if we have a list of new features and fixes of the original game bugs. So visitors could clearly understand the amount of work we've already done in this project.
I'm concerning on the form of this list... It could probably look like our change logs from github wiki, but I'd like to provide a screenshots to the most of new features, describing each improvement there.
I'm still in a process of creating a full list, but I can already provide a few dozens of screenshots of the improvements. Later I'll add even more of them.

**Reply by Branikolog** [reply]
2-hex creature has 2-hex cursor shadow while pointing a battlefield.
![2-hex movement](https://user-images.githubusercontent.com/55348946/126502063-1e4c9858-3d1b-4730-ae9d-bd2a4af61532.png)

Cursor's counter is increased up to 7+ days. (Originally counter has 4+ days)
![7 day cursor](https://user-images.githubusercontent.com/55348946/126502326-969a5c10-06fd-4310-86eb-3f1af6fda00a.png)

Battle speed 10-step adjustment. (Originally adjustment has only 3 steps)
![Battle speed 10-step adjustment](https://user-images.githubusercontent.com/55348946/126502620-0e2d50ae-3e2f-442a-9b64-8df60321801a.png)

Hero moving 10-step adjustment. (Originally adjustment has only 4 steps)
![Hero Speed 10 step adjustment](https://user-images.githubusercontent.com/55348946/126504959-873b052e-2d10-40ed-98e8-33774b958cc5.png)

Fixed dying cavalry sprite. (Originally sprite has partly wrong colors)
![Cavalry sprite fix](https://user-images.githubusercontent.com/55348946/126505042-234255c9-bfe5-425c-b6ec-6838c6886e66.png)

Full hero info could be checked on clicking a portrait during battle.
![Check hero info during battle](https://user-images.githubusercontent.com/55348946/126505353-497ebea2-c703-445e-88bd-9bb96a387832.png)

Arrows in a meeting heroes window were turned into a buttons, allowing to move all artifacts or creatures from one hero to another.
![Fast move buttons](https://user-images.githubusercontent.com/55348946/126506372-9de06845-3eef-43fa-99c0-119869f8a2df.png)

Selected creature or artifact in hero meeting window has red glowing frame. (Originally the selection has white static frame)
![Glowing selection of artifacts   creatures](https://user-images.githubusercontent.com/55348946/126506545-39ca175b-d4cc-4061-8eda-48c115abe292.png)

Reworked town portal window.
![New Town Portal](https://user-images.githubusercontent.com/55348946/126506806-8581296c-9ae6-4a67-868c-7476ab8e24bb.png)

Well has animated creature sprites. (Originally they were static)
![Well Animation](https://user-images.githubusercontent.com/55348946/126506941-5af8ee17-a57e-4838-92b8-7f6454f91727.png)

Creatures could be hired in a well.
![Well Recruit](https://user-images.githubusercontent.com/55348946/126507796-dc63091b-5677-43d4-a9c2-9c9db66db931.png)

Well MAX button allows to hire all available creatures.
![Well  hire button](https://user-images.githubusercontent.com/55348946/126510312-0e116539-6321-4344-9378-f959bc873bcb.png)

Selection of upgraded or non-upgraded creatures to hire.
![Upgrade switch](https://user-images.githubusercontent.com/55348946/126507905-ac1717e3-2b18-4d6f-b8d9-6fec1cb4aeba.png)

Maximum quantity of hireable creatures are selected on default. (Originally hiring counter is set to 0)
![Max hire on default](https://user-images.githubusercontent.com/55348946/126508157-a67a2df8-4c9d-4e2c-a1f5-2a3a3e1b9ee4.png)

MAX button has inactive state in empty dwelling.
![Inactive Max hire button](https://user-images.githubusercontent.com/55348946/126509786-5ab96741-73ca-4ee3-8492-f052dd4a0d32.png)

MIN hire button.
![MIN hire button](https://user-images.githubusercontent.com/55348946/126510069-f87e5d5c-6b5f-450c-9b4b-b386619e1683.png)

Larger Creatures' counters.
![Large creature counters - Copy](https://user-images.githubusercontent.com/55348946/126510476-371de5fe-ae7f-4227-89e5-d2357db2c30e.png)

Area-based attacking spells have highlighted hexes. 
![Spell area](https://user-images.githubusercontent.com/55348946/126510653-10d71856-56d8-4583-bd84-d8baf9419125.png)

Lich shooting area is highlighted on a battlefield
![Lich Shoot Area](https://user-images.githubusercontent.com/55348946/126510815-09b9807c-2e19-45b9-9a10-022f912c3a90.png)

**Reply by Branikolog** [reply]
Kingdom income window on clicking the resource box in castle window.
![Income window](https://user-images.githubusercontent.com/55348946/126789522-f7ad70a4-3d1b-4ae3-8c07-200b0c696ee4.png)

Instant Battle option.
![Instant Battle](https://user-images.githubusercontent.com/55348946/126790450-68d88a6f-4bb4-49a0-a69e-0b332efa1675.png)

Restart instant battle manually.
![Restart instant battle](https://user-images.githubusercontent.com/55348946/126790547-e23331ab-e70c-4400-a48a-733d57fc941a.png)

Yellow heading of recruit window.
![Yellow heading in recruit window](https://user-images.githubusercontent.com/55348946/126790714-be4dcc87-23a6-4bb1-a835-c19030b72b42.png)

**Reply by qarkai** [reply]
Monster abilities ![](https://i.servimg.com/u/f85/20/25/62/08/bone_d10.png)

**Reply by Branikolog** [reply]
Mage guilds have centered sprites, while viewing scrolls.
![Barbarian mage guild](https://user-images.githubusercontent.com/55348946/126899703-42c1c960-9767-4f77-a1f8-851b15e1d810.png)
![Knight mage guild](https://user-images.githubusercontent.com/55348946/126899704-b1377e27-fb2a-4fe0-9064-b610374ab2c4.png)
![Necromancer mage guild](https://user-images.githubusercontent.com/55348946/126899708-3f8fcf5b-447b-4f5e-ac34-99c9de781d4a.png)
![Sorceress mage guild](https://user-images.githubusercontent.com/55348946/126899710-015af3c8-7452-42ca-8cfe-a52009e5e2bb.png)
![Warlock mage guild](https://user-images.githubusercontent.com/55348946/126899713-f036e241-f533-458c-b13b-f731b9747d64.png)
![Wizard mage guild](https://user-images.githubusercontent.com/55348946/126899715-18c01e3e-9213-4daf-8277-1e0a1899a5dc.png)

**Reply by Branikolog** [reply]
Hero army could be reorganized for joining creatures via special button.
![Hero Info button](https://user-images.githubusercontent.com/55348946/130244139-c9ec342f-5337-4db5-8913-169d3070b5ae.png)

Hero info could be checked while choosing new skills on level up. 
![Hero Info button 2](https://user-images.githubusercontent.com/55348946/130244350-ee30243b-a7e3-47c5-b1af-9234eeb46a15.png)

Available gold and needed experience for level up are shown while obtaining treasure chest.
![Treasure chest info](https://user-images.githubusercontent.com/55348946/130244531-00f02029-b389-4d55-9ec2-9b422d12d254.png)

Unavailable buildings are shown in gray color (H3-like).
![Grayed building](https://user-images.githubusercontent.com/55348946/130244767-06d4e92b-ef65-4bfc-a01e-64386531f9d4.png)

Visited and unvisited status of multiple objects on adventure map.
![Visited info](https://user-images.githubusercontent.com/55348946/130245000-2abe3f1c-4db6-4ed8-8bb2-15288c405930.png)

Once visited, Witch Hut indicates the skill, could be obtained.
![Withc hut skill](https://user-images.githubusercontent.com/55348946/130245172-0321660e-6b88-4c6f-a9d4-179ff66231f4.png)

Magic Shrines show the spell, that could be learned inside after visiting.
![Shrine Spell](https://user-images.githubusercontent.com/55348946/130245295-846ac74d-c309-464f-a94f-1a80067f4dec.png)

**Reply by Branikolog** [reply]
Hero info window shows disabled "Dismiss" button in a castle plus info that explains why player cannot dismiss a hero.
![Disabled Dismiss button](https://user-images.githubusercontent.com/55348946/131219892-698e62d9-c817-493b-8633-b5e051d86b09.png)

Marketplace button allows hero exchange some resources during creature joining dialog.
![Marketplace button](https://user-images.githubusercontent.com/55348946/131220046-00a836fc-13bc-484d-b871-c5e5409a06ef.png)

A tower of Knight Castle no more overlaps a flag on a main gates.
![Flag over tower](https://user-images.githubusercontent.com/55348946/131220079-96ebfae7-fd78-4981-9cb9-f63b5dd894d8.png)

Status information of shipyard changed from "Dock" to "Shipyard".
![Dock-Shipyard](https://user-images.githubusercontent.com/55348946/131220099-4e45f4eb-bfba-43c1-9393-53d7f61e8164.png)

All info windows in game are placed exactly in a center of the screen.
![Centered all info windows](https://user-images.githubusercontent.com/55348946/131220128-88ca3dee-7f8f-42e4-9089-3b0b15f9c19a.png)

During castle selection a "Spell Cast" button is in a disabled state.
![Disabled cast spell button for castles](https://user-images.githubusercontent.com/55348946/131220150-3e6595bc-b445-409a-94a9-04b88ec96209.png)

Capital letters are used for creatures names in a status bar.
![Capital letters for creatures names in castle status bar](https://user-images.githubusercontent.com/55348946/131220174-bdb75852-b133-43e0-b3d5-19df57cdffcf.png)

**Reply by Branikolog** [reply]
Save files have creation date.
![Save file date](https://user-images.githubusercontent.com/55348946/131251292-09a3157d-be6f-4945-baa6-9a87096f3272.png)

Optional Army order boxes.
![Army Order](https://user-images.githubusercontent.com/55348946/131251308-14a12768-8535-4279-baf8-7a8783888222.png)

Morale modifiers in hero info window show the name of each grade. (While in the original there were only "Good", "Bad" or "Neutral"
![Morale Modifiers in hero info](https://user-images.githubusercontent.com/55348946/131251335-e16a79e1-5321-4d60-8288-86aed119cb16.png)

**Reply by Branikolog** [reply]
Fast creature splitting into several equal stacks.
![Fast splitting](https://user-images.githubusercontent.com/55348946/137479684-ddcd76f9-8207-4b43-9708-05b739dcef9d.png)

Higher resolution support.
![image](https://user-images.githubusercontent.com/55348946/137479981-a83317bd-9579-4f85-beb3-fa440c5428f8.png)

Game setting via "config" button in main menu.
![image](https://user-images.githubusercontent.com/55348946/137480175-911b4eff-3212-4e6b-93b8-2e0b3a354c85.png)

If creature inflicts damage on two hexes they are both shadowed. 
![2-hex attack highlight](https://user-images.githubusercontent.com/55348946/137480764-09818c7d-eaf6-4d77-867c-54666b468149.png)

Each map in the list has PoL or SW label.
![Map type Label](https://user-images.githubusercontent.com/55348946/137480984-6004beda-842c-4dc5-bd7c-cb5ad5b740af.png)

Active spell duration in creature characteristics window during battle.
![Spell Duration](https://user-images.githubusercontent.com/55348946/137481793-25a3b562-8aaf-4db1-8aba-3fd144fb2292.png)

Spell description in creature characteristics window during battle.
![Spell description](https://user-images.githubusercontent.com/55348946/137481922-fab118b2-2e13-4527-aac9-271e5ee7cc16.png)

Save file info.
![Save file  Info](https://user-images.githubusercontent.com/55348946/137483488-f9a4c630-a1bf-43c6-b005-3fb74a892796.png)

Captain Quarters in Necromancers town is drawn without bridge to nowhere.
![Captain Quarters double](https://user-images.githubusercontent.com/55348946/137484484-2feca19f-cb1b-4741-bb5d-5b40eec141fa.png)

Added missing part to a farm in Knight castle.
![Farm double](https://user-images.githubusercontent.com/55348946/137484595-7423a9f8-6e6f-45f4-9aee-65ede5856d97.png)

**Reply by Branikolog** [reply]
Added missing shadow for a spell book.
![Book Shadow](https://user-images.githubusercontent.com/55348946/137690285-4d82da92-2678-48ac-b24c-99a8a2cae040.png)

Freeman's Foundry window shows all upgraded types of creatures. (Originally only two creatures' icons fit the window.)
![freeman's foundry](https://user-images.githubusercontent.com/55348946/137693081-eb892614-6db3-47e0-9db4-ed94d8e41d82.png)

**Comment by oleg-derevenetz** [comment]
Hi @Branikolog I added an introductory description in this way:

https://oleg-derevenetz.github.io/fheroes2/

I suppose there is already A LOT of things works fine enough, so there is just no need to use promises in the future tense to draw attention to the project, let's leave that to some kind of endless game project that has nothing but screenshots, preprocessed videos and promises (I think anyone can give an example of a couple of such 😊). All we need to do is just present the already existing game features in a flashy way. I have an idea to make some screenshots from the OG and from the fheroes2 in similar situations to show them side-by-side ("before and now") to demonstrate some of the UI improvements. Also we may advertise some game mechanics that is just absent in OG - such as showing turn order in battle, castle guardians, maybe something else.

@ihhub How do you think, are screenshots from OG on fheroes2 website are fine in terms of potential copyright issues?

**Reply by ihhub** [reply]
Hi @oleg-derevenetz , personally I do not know the impact of placing screenshots from OG on the website so I can say that we should try to avoid unless it's unavoidable. Some features is hard to show on screenshots so we'll need video for this.

**Reply by ihhub** [reply]
@Branikolog , small remarks about screenshots:
- town portal doesn't need screenshots from OG
- the arrow in hero's meeting dialog should be pressed
- remove larger creature's counts (not worth to mention)

**Reply by Branikolog** [reply]
@oleg-derevenetz 
Everything would be better than nothing, we had before. ;)

@ihhub 
I think everything is worth to mention. ) I even added to my list some pixelfixes of buildings in castles. Larger counts - is nice improvement, as in the OG they were too tiny and in future, at mobile devices this will really help users to distinguish these counts.
Also, why shouldn't we show the original "Town Portal" implementation? Lots of people trying fheroes2 could never played the original game so they would even don't know, if this spell had such a primitive UI. 

edit: Arrow in hero meeting window IS pressed. ;)

**Reply by ihhub** [reply]
Hi @oleg-derevenetz  and @Branikolog , let's go with adding the description of the project first as it's an obvious change.

**Comment by ihhub** [comment]
As discussed I've created a new wiki page [Features-and-enhancement-of-the-project](https://github.com/ihhub/fheroes2/wiki/Features-and-enhancement-of-the-project) where @Branikolog  will populate with all these images and brief description. This is needed to avoid images being a part of git as they will occupy a lot of space. Once the page will be population a little we can add a separate *.md file to be used for generated website.
